Impro 1669

Kat-90 opened the pull request:
Addresses #GitHubissuenum

Description:
Updated units tests to remove set_up_cube() function and replace with function within the synthetic_data plugin.

Testing:
 - [X] Ran tests and they passed OK
 - [ ] Added new tests for the new feature(s)

CLA
 - [ ] If a new developer, signed up to CLA

benfitzpatrick commented:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Here is an overview of what got changed by this pull request:


```diff

Issues
======
- Added 6
           

Clones added
============
- improver_tests/nbhood/nbhood/test_BaseNeighbourhoodProcessing.py  4
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/metoppv_tech/improver/pullRequest?prid=6164254&bid=20133112)

benfitzpatrick commented on improver_tests/nbhood/nbhood/test_BaseNeighbourhoodProcessing.py:
```diff
@@ -531,32 +502,48 @@ def test_radii_varying_with_lead_time(self):
+            cube, 
+            coord_points=time_points, 
+            coord_name="time", 
+            is_datetime="true", 
```
Codacy found an issue: [Trailing whitespace](https://app.codacy.com/manual/metoppv_tech/improver/pullRequest?prid=6164254)

benfitzpatrick commented on improver_tests/nbhood/nbhood/test_BaseNeighbourhoodProcessing.py:
```diff
@@ -531,32 +502,48 @@ def test_radii_varying_with_lead_time(self):
+        )
+        time_points = [datetime(2017, 11, 10, 2), datetime(2017, 11, 10, 3), datetime(2017, 11, 10, 4)]
+        cube = add_coordinate(
+            cube, 
```
Codacy found an issue: [Trailing whitespace](https://app.codacy.com/manual/metoppv_tech/improver/pullRequest?prid=6164254)

benfitzpatrick commented on improver_tests/nbhood/nbhood/test_BaseNeighbourhoodProcessing.py:
```diff
@@ -531,32 +502,48 @@ def test_radii_varying_with_lead_time(self):
+        time_points = [datetime(2017, 11, 10, 2), datetime(2017, 11, 10, 3), datetime(2017, 11, 10, 4)]
+        cube = add_coordinate(
+            cube, 
+            coord_points=time_points, 
```
Codacy found an issue: [Trailing whitespace](https://app.codacy.com/manual/metoppv_tech/improver/pullRequest?prid=6164254)

benfitzpatrick commented on improver_tests/nbhood/nbhood/test_BaseNeighbourhoodProcessing.py:
```diff
@@ -531,32 +502,48 @@ def test_radii_varying_with_lead_time(self):
+        cube = add_coordinate(
+            cube, 
+            coord_points=time_points, 
+            coord_name="time", 
```
Codacy found an issue: [Trailing whitespace](https://app.codacy.com/manual/metoppv_tech/improver/pullRequest?prid=6164254)

benfitzpatrick commented on improver_tests/nbhood/nbhood/test_BaseNeighbourhoodProcessing.py:
```diff
@@ -44,7 +45,12 @@
+    set_up_variable_cube,
+    add_coordinate,
+)
+from datetime import datetime
```
Codacy found an issue: [standard import "from datetime import datetime" should be placed before "import iris"](https://app.codacy.com/manual/metoppv_tech/improver/pullRequest?prid=6164254)

benfitzpatrick replied:
Codacy found an issue: [standard import "from datetime import datetime" should be placed before "import iris"](https://app.codacy.com/manual/metoppv_tech/improver/pullRequest?prid=6164254)

benfitzpatrick commented on improver_tests/nbhood/nbhood/test_BaseNeighbourhoodProcessing.py:
```diff
@@ -744,19 +768,18 @@ def test_use_mask_cube_occurrences_masked(self):
-        )
-        cube = set_up_cube(
-            zero_point_indices=((0, 0, 2, 2),), num_grid_points=5, num_time_points=1
+        
```
Codacy found an issue: [Trailing whitespace](https://app.codacy.com/manual/metoppv_tech/improver/pullRequest?prid=6164254)

benfitzpatrick commented:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Here is an overview of what got changed by this pull request:


```diff

Issues
======
- Added 1
           

Clones added
============
- improver_tests/nbhood/nbhood/test_BaseNeighbourhoodProcessing.py  4
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/metoppv_tech/improver/pullRequest?prid=6164254&bid=20133112)

benfitzpatrick commented:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Here is an overview of what got changed by this pull request:


```diff

Clones added
============
- improver_tests/nbhood/nbhood/test_BaseNeighbourhoodProcessing.py  4
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/metoppv_tech/improver/pullRequest?prid=6164254&bid=20133112)

cgsandford commented on improver_tests/nbhood/nbhood/test_BaseNeighbourhoodProcessing.py:
```diff
@@ -785,10 +804,11 @@ def test_use_mask_cube_occurrences_masked_irregular(self):
                 [1.000000, 0.000000, 1.000000, 0.000000, 0.000000],
             ]
         )
```
Don't think this statement has any effect - you already set up the cube with this data (as you tidied up, above).

Kat-90 replied:
Need to clarify which statement you're referring to here, it looks like the expected array in  def test_use_mask_cube_occurrences_masked_irregular, but I don't think it is because that array is necessary for the expected data.

cgsandford replied:
It was about the masked data - but I misread it, sorry.  So please ignore the original comment.

cgsandford reviewed (changes requested):
Pretty good.  Main comments relate to test names, and a few areas where further tidy up / reduced duplication would be beneficial.  But on the whole, nice work.

benfitzpatrick commented:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Here is an overview of what got changed by this pull request:


```diff

Clones added
============
- improver_tests/nbhood/nbhood/test_BaseNeighbourhoodProcessing.py  4
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/metoppv_tech/improver/pullRequest?prid=6164254&bid=20133112)

Kat-90 commented on improver_tests/nbhood/nbhood/test_BaseNeighbourhoodProcessing.py:
```diff
@@ -44,6 +45,11 @@
+    set_up_variable_cube,
+)
 
 from ...calibration.ensemble_calibration.helper_functions import (
     add_forecast_reference_time_and_forecast_period,
```
Done.

Kat-90 commented on improver_tests/nbhood/nbhood/test_BaseNeighbourhoodProcessing.py:
```diff
@@ -491,36 +449,45 @@ def test_neighbourhood_method_does_not_exist(self):
             neighbourhood_method = CircularNeighbourhood
             NBHood(neighbourhood_method, self.RADIUS)(self.cube)
 
     def test_multiple_realizations(self):
```
Done.

Kat-90 commented on improver_tests/nbhood/nbhood/test_BaseNeighbourhoodProcessing.py:
```diff
@@ -491,36 +449,45 @@ def test_neighbourhood_method_does_not_exist(self):
 
     def test_no_realizations(self):
-        """Test when the array has no realization coord."""
-        cube = set_up_cube_with_no_realizations()
+        """Test when the array has no threshold coord."""
```
Done.

Kat-90 commented on improver_tests/nbhood/nbhood/test_BaseNeighbourhoodProcessing.py:
```diff
@@ -571,10 +556,23 @@ def test_radii_varying_with_lead_time_check_data(self):
         Test that the expected data is produced when the radius
         varies with lead time.
         """
```
I've removed the first one and added the extra assert into the last one, but just wanted to check that we definitely don't want to check that the fp coord isn't changed in that middle test, before removing it.

cgsandford replied:
There would be no logical reason for this code to touch the forecast period coordinate - so I don't think we need that check, no.

Kat-90 replied:
Done.

Kat-90 commented on improver_tests/nbhood/nbhood/test_BaseNeighbourhoodProcessing.py:
```diff
@@ -610,18 +602,29 @@ def test_radii_varying_with_lead_time_check_data(self):
         1. The radius varies wtih lead time.
-        2. The cube contains multiple realizations."""
-        cube = set_up_cube(
-            zero_point_indices=((1, 0, 7, 7), (1, 1, 7, 7), (1, 2, 7, 7)),
-            num_time_points=3,
-            num_realization_points=2,
+        2. The cube contains multiple thresholds."""
```
Done.

Kat-90 commented on improver_tests/nbhood/nbhood/test_BaseNeighbourhoodProcessing.py:
```diff
@@ -610,18 +602,29 @@ def test_radii_varying_with_lead_time_check_data(self):
-            num_realization_points=2,
+        2. The cube contains multiple thresholds."""
+
+        data = np.ones((2, 16, 16), dtype=np.float32)
+        data[1, 7, 7] = 0
+        cube = set_up_probability_cube(
+            data, thresholds=[278, 279], spatial_grid="equalarea",
+        )
+
+        time_points = [
+            datetime(2017, 11, 10, 2),
+            datetime(2017, 11, 10, 3),
+            datetime(2017, 11, 10, 4),
+        ]
+        cube = add_coordinate(
+            cube,
+            coord_points=time_points,
+            coord_name="time",
+            is_datetime="true",
+            order=[1, 0, 2, 3],
         )
```
Done.

Kat-90 commented:
I have addressed your comments, and updated some more of the code in test_BaseNeighbourhoodProcessing.py, but there are some clarifications before it is complete. I will push the new version and wait for you to come back next week.
Thanks

benfitzpatrick commented:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Here is an overview of what got changed by this pull request:


```diff

Clones removed
==============
+ improver_tests/nbhood/nbhood/test_BaseNeighbourhoodProcessing.py  -16
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/metoppv_tech/improver/pullRequest?prid=6164254&bid=20133112)

cgsandford commented on improver_tests/nbhood/nbhood/test_BaseNeighbourhoodProcessing.py:
```diff
@@ -721,10 +666,11 @@ def test_use_mask_cube_occurrences_not_masked(self):
+        cube = set_up_variable_cube(data, spatial_grid="equalarea",)
+
         mask_cube = cube.copy()
         mask_cube.data = np.array(
```
Nice-to-have - you can set this up more concisely as:
```
mask_cube = cube.copy(data=np.ones((5, 5), dtype=np.float32))
```

Kat-90 replied:
Done.

cgsandford reviewed (approved):
Nice work 👍

benfitzpatrick commented:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Here is an overview of what got changed by this pull request:


```diff

Clones removed
==============
+ improver_tests/nbhood/nbhood/test_BaseNeighbourhoodProcessing.py  -20
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/metoppv_tech/improver/pullRequest?prid=6164254&bid=20133112)

benfitzpatrick commented:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Here is an overview of what got changed by this pull request:


```diff

Issues
======
+ Solved 1
           

Complexity decreasing per file
==============================
+ improver_tests/nbhood/nbhood/test_BaseNeighbourhoodProcessing.py  -3
         

Clones removed
==============
+ improver_tests/nbhood/nbhood/test_BaseNeighbourhoodProcessing.py  -22
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/metoppv_tech/improver/pullRequest?prid=6164254&bid=20133112)

benfitzpatrick commented:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Here is an overview of what got changed by this pull request:


```diff

Issues
======
+ Solved 1
- Added 1
           

Complexity decreasing per file
==============================
+ improver_tests/nbhood/nbhood/test_BaseNeighbourhoodProcessing.py  -3
         

Clones removed
==============
+ improver_tests/nbhood/nbhood/test_BaseNeighbourhoodProcessing.py  -22
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/metoppv_tech/improver/pullRequest?prid=6164254&bid=20133112)

benfitzpatrick commented on improver_tests/nbhood/nbhood/test_BaseNeighbourhoodProcessing.py:
```diff
@@ -672,11 +451,8 @@ def test_radii_varying_with_lead_time_with_interpolation_check_data(self):
-        )
-        expected = np.ones_like(cube.data)
+
+        expected = np.ones_like(multi_time_cube.data)
```
Codacy found an issue: [Undefined variable 'multi_time_cube'](https://app.codacy.com/manual/metoppv_tech/improver/pullRequest?prid=6164254)

benfitzpatrick commented:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Here is an overview of what got changed by this pull request:


```diff

Issues
======
+ Solved 1
           

Complexity decreasing per file
==============================
+ improver_tests/nbhood/nbhood/test_BaseNeighbourhoodProcessing.py  -3
         

Clones removed
==============
+ improver_tests/nbhood/nbhood/test_BaseNeighbourhoodProcessing.py  -22
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/metoppv_tech/improver/pullRequest?prid=6164254&bid=20133112)

MoseleyS reviewed (changes requested):
Please could you tidy up the use of the threshold coord in Test__process. self.cube is defined with a length-1 threshold coord. test_multiple_thresholds defines a new cube with four thresholds. test_2d slices the length-1 threshold coord.
I think it would be clearer to define a 2D cube in the first place (achievable with the current definition, but add `self.cube = iris.util.squeeze(self.cube)` - this has the benefit of keeping the threshold definition as a scalar coord) which means you can remove test_2d. You will have to tweak a few places that refer to three or four dimensions, as you will now only have two or three. (hint: For definition of `self.multi_time_cube`, you can remove the `order` keyword).

MoseleyS commented:
> 
> 
> I think it would be clearer to define a 2D cube in the first place

On second thoughts, there's no reason to retain the threshold scalar coord on `self.cube`, so it would be cleaner still to just define a simpler 2D cube in the first place.

benfitzpatrick commented:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Here is an overview of what got changed by this pull request:


```diff

Issues
======
+ Solved 1
           

Complexity decreasing per file
==============================
+ improver_tests/nbhood/nbhood/test_BaseNeighbourhoodProcessing.py  -3
         

Clones removed
==============
+ improver_tests/nbhood/nbhood/test_BaseNeighbourhoodProcessing.py  -22
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/metoppv_tech/improver/pullRequest?prid=6164254&bid=20133112)

MoseleyS reviewed (approved):
Looks good now. Thanks Kat.
